src/Core/System/Snippet/BundleSnippetValidator.php

```diff
@@ -0,0 +1,169 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Snippet;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\Snippet\Files\GenericSnippetFile;
+use Shopware\Core\System\Snippet\Files\SnippetFileCollection;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+#[Package('discovery')]
+class BundleSnippetValidator implements BundleSnippetValidatorInterface
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly SnippetFileCollection $deprecatedSnippetFiles,
+        private readonly SnippetFileHandler $snippetFileHandler,
+        private readonly string $projectDir,
+        private readonly KernelInterface $kernel
+    ) {
+    }
+
+    public function validateForBundles(array $bundles = []): array
+    {
+        $files = $this->getFileFromBundles($bundles);
+
+        $snippetFileMappings = [];
+        $availableISOs = [];
+        foreach ($files as $snippetFile) {
+            $availableISOs[] = $snippetFile->getIso();
+
+            if (!\array_key_exists($snippetFile->getIso(), $snippetFileMappings)) {
+                $snippetFileMappings[$snippetFile->getIso()] = [];
+            }
+
+            $json = $this->snippetFileHandler->openJsonFile($snippetFile->getPath());
+
+            foreach ($this->getRecursiveArrayKeys($json) as $keyValue) {
+                $snippetFileMappings[$snippetFile->getIso()][key($keyValue)] = [
+                    'path' => str_ireplace($this->projectDir, '', $snippetFile->getPath()),
+                    'availableValue' => array_shift($keyValue),
+                ];
+            }
+        }
+
+        return $this->findMissingSnippets($snippetFileMappings, $availableISOs);
+    }
+
+    /**
+     * @param array<int, string> $bundles
+     */
+    protected function getFileFromBundles(array $bundles = []): SnippetFileCollection
```

Comment:
does this method have to be protected? I would suggest to make it private, to lower the public API. as we are introducing this completely new